File: plugin/shared/src/main/java/com/jonathanpotts/cartographytable/shared/models/BlockModel.java
package com.jonathanpotts.cartographytable.shared.models;

import java.util.Map;

/**
 * Stores data used to process a block.
 */
public class BlockModel {
    /**
     * Material ID ordinal.
     */
    public int mat;

    /**
     * Additional block data.
     */
    public String data;

    /**
     * Map containing lighting data.
     */
    public Map<Integer, Integer> light;
}

File: plugin/shared/src/main/kotlin/com/jonathanpotts/cartographytable/shared/models/BlockModel.kt
package com.jonathanpotts.cartographytable.shared.models

import kotlinx.serialization.Serializable

/**
 * Stores data used to process a block.
 *
 * @property mat Material ID ordinal.
 * @property data Additional block data.
 * @property light Map containing lighting data.
 */
@Serializable
data class BlockModel(
    var mat: Int = -1,
    var data: String? = null,
    var light: MutableMap<Int, Int> = mutableMapOf()
)
